streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamWindowReduce.java

```diff
@@ -78,45 +77,20 @@ public void process(K key, V value) {
 
             Map<Long, W> matchedWindows = windows.windowsFor(timestamp);
 
-            long timeFrom = Long.MAX_VALUE;
-            long timeTo = Long.MIN_VALUE;
+            // try update the window, and create the new window for the rest of unmatched window that do not exist yet
+            for (Map.Entry<Long, W> entry : matchedWindows.entrySet()) {
+                V oldAgg = windowStore.fetch(key, entry.getKey());
+                V newAgg = oldAgg;
```

Comment:
don't need to set `oldAgg`, will be overwritten in following if-else anyway (cf. comment below)